httplib.h

```diff
@@ -454,11 +454,50 @@ class ContentReader {
   MultipartReader multipart_reader_;
 };
 
+enum RequestMethod {
+  UNKNOWN = -1,
+  GET = 0,
+  HEAD,
+  POST,
+  PUT,
+  DEL,
+  CONNECT,
+  OPTIONS,
+  TRACE,
+  PATCH,
+  PRI,
+};
+
+namespace detail {
+  RequestMethod string_to_method(const std::string& str) {
+    static const std::unordered_map<std::string, int> method_map{
+          {"GET", GET},     {"HEAD", HEAD},       {"POST", POST},  {"PUT", PUT},   
+          {"DELETE", DEL},  {"CONNECT", CONNECT}, {"OPTIONS", OPTIONS}, 
+          {"TRACE", TRACE}, {"PATCH", PATCH},     {"PRI", PRI}};
+    
+    auto it = method_map.find(str);
+    if (it != method_map.end()) {
+      return static_cast<RequestMethod>(it->second);
+    }
+
+    return UNKNOWN;
+  }
+
+  std::string method_to_string(RequestMethod method) {
+    static const std::vector<std::string> method_string{
+        "GET",     "HEAD",       "POST",  "PUT",   "DELETE", "CONNECT", 
+        "OPTIONS", "TRACE",      "PATCH", "PRI"};
+
+    if (method != UNKNOWN) { return method_string[method]; }
+    return std::string();
+  }
+}
+
 using Range = std::pair<ssize_t, ssize_t>;
 using Ranges = std::vector<Range>;
 
 struct Request {
-  std::string method;
+  RequestMethod method = UNKNOWN;
```

Comment:
This is a breaking change and will cause build errors of existing users's code using `send` method. Could you change back `method` to `std::string` to `RequestMethod` please?

Reply:
oh ， I'll revert this change.

Reply:
reverted the breaking chage.